conftest.py

```diff
@@ -57,6 +59,16 @@ def get_software():
             )
 
 
+@pytest.fixture(scope="session")
+def event_loop():
+    if threading.current_thread() is not threading.main_thread():
+        os.exit(1)
```

Comment:
I added this as a workaround based on a suggestion from @graingert. All tests passed on the most recent commit, I'm going to run tests a few more times to make sure things work consistently (previous failures only happened occasionally)